fluent-validator/src/test/java/com/baidu/unbiz/fluentvalidator/FluentValidatorWIthoutAnnotationsTest.java

```diff
@@ -0,0 +1,111 @@
+        }
+    }
+
+    class SeatBeltValidator extends ValidatorHandler<Boolean> implements Validator<Boolean> {
+        private String name;
+
+        public SeatBeltValidator(String name) {
+            this.name = name;
+        }
+
+        public boolean validate(ValidatorContext context, Boolean hasSeatbeltOn) {
+            if (!hasSeatbeltOn) {
+                ValidationError error = new ValidationError();
+                error.setErrorCode(123);
+                error.setErrorMsg("passenger " + name + " must have their seat belt on");
+                context.addError(error);
+            }
+
+            return hasSeatbeltOn;
+        }
+    }
+
+    class SeatBeltValidatorWithoutError extends ValidatorHandler<Boolean> implements Validator<Boolean> {
+        public boolean validate(ValidatorContext context, Boolean hasSeatbeltOn) {
+            return hasSeatbeltOn;
+        }
+    }
+
+    class PassengerValidator extends ValidatorHandler<Passenger> implements Validator<Passenger> {
+        @Override
+        public boolean validate(ValidatorContext context, Passenger passenger) {
+            ComplexResult result = FluentValidator
+                    .checkAll()
+                    .on(passenger.hasSeatbeltOn, new SeatBeltValidator(passenger.name))
+                    .on(passenger.hasSeatbeltOn, new SeatBeltValidatorWithoutError())
+                    .failOver()
+                    .doValidate()
+                    .result(ResultCollectors.toComplex());
+
+            for (ValidationError error : result.getErrors()) {
+                context.addError(error);
+            }
+
+            return result.isSuccess();
+        }
+    }
+
+    class CarValidator extends ValidatorHandler<Car> implements Validator<Car> {
+        @Override
+        public boolean validate(ValidatorContext context, Car car) {
+            ComplexResult result = FluentValidator
+                    .checkAll()
+                    .onEach(car.passengers, new PassengerValidator())
+                    .failOver()
+                    .doValidate()
+                    .result(ResultCollectors.toComplex());
+
+            for (ValidationError error : result.getErrors()) {
+                context.addError(error);
+            }
+
+            return result.isSuccess();
+        }
+    }
+
+    @Test
+    public void itCanValidateComplexObjects() {
+        Car car = new Car();
+        car.addPassenger(new Passenger("Adam", false));
+        car.addPassenger(new Passenger("Joe", true));
+
+        ComplexResult result = FluentValidator
+                .checkAll()
+                .on(car, new CarValidator())
+                .failOver()
+                .doValidate()
+                .result(ResultCollectors.toComplex());
+
+        assertThat(result.getErrors().size(), equalTo(1));
+        assertThat(result.getErrors().get(0).getErrorMsg(), equalTo("passenger Adam must have their seat belt on"));
+    }
+}
```

Comment:
The code duplication within each model validator (`Car` and `Passenger`) led us to the other pull request that we've submitted.

This section of code is unnecessarily repeated each time:
```FluentValidator.checkAll()```
and 
```.failOver().doValidate().result(ResultCollectors.toComplex());```

the only part that varies each time are the `#on` and `#onEach` method calls to the fluent validator.